ServerCore/Pages/Submissions/Index.cshtml

```diff
@@ -459,16 +459,30 @@
 }
 else
 {
+    string cssFileName = String.IsNullOrEmpty(Model.Puzzle.CustomCSSFile) ? "" : Model.Puzzle.CustomCSSFile.Trim();
+    @if (cssFileName.Length > 0)
+    {
+        // Add class to body wrapper here so only this page's content gets it, not every page
+        <script type="text/javascript">
+            document.addEventListener("DOMContentLoaded", () => {
+                let bodyWrapper = document.querySelector(".body-content");
+                bodyWrapper.classList.add("body-content-customizable");
+            });
+        </script>
+        cssFileName += (!cssFileName.EndsWith(".css")) ? ".css" : "";
+        <link rel="stylesheet" href="~/css/@(Model.Event.HomePartial)/@cssFileName" />
```

Comment:
Is there a requirement to load from this server? If authors could upload css as materials, this could be self-service (particularly useful for module hunt)

Reply:
As implied by my huge comment above, this is going to change to indeed load from the resources folder for all future events so that absolutely nothing for future events will come from the wwwroot folder.

Reply:
I think it's quite reasonable/desirable to allow for custom CSS per puzzle, so that a puzzle in a module event can change whatever it likes if desired. No problem if the default is to go to the resources folder, so long as any puzzle that wants to do something custom can put that CSS file in its materials and provide a URL of, say, "../mypuzzle/mycustom.css".

Reply:
I think we're getting our wires crossed here a little bit.  CSS theming for the puzzle itself will ideally always go alongside the puzzle materials that are uploaded per-puzzle and be addressed relative to the puzzle file, assuming it's uploaded correctly.  This PR won't change anything about that since we're already able to do exactly that.  

This PR is specifically for theming the _host_ page so that the entire thing can be themed, and not just inside the puzzle's iframe, in case that's something that's desired.  In this case, you _could_ use the same file to theme both the host page and the puzzle itself, since the puzzle should _also_ have access to everything in the resources folder, whereas the host page _only_ has access to the resources folder.  

The way I'm envisioning this being used this year is that each year's puzzle(s) will have a different look, so each host page will need a different theme depending on the puzzle.  This definitely will be useful for module hunts too; I just want to make sure we're clear on the boundary between files for the website itself, and files for a puzzle.  There's currently nothing _enforcing_ that boundary, which is why if we get it confused now, it will only get worse over time.

Reply:
For module events with no true admin team, a puzzle may want to modify top-level CSS for their puzzle. They shouldn't have to use global resources because they may not have access.

I don't think this is the most common use case, but might as well light it up along with the rest.

Reply:
Ah, I see what you're saying, you want the host page to be able to load css from more than JUST the resources folder.  That makes sense, I hadn't thought about the fact that only admins (and authors?) can access that.  That seems reasonable, so how about this:  if you ONLY supply a file name, it will be assumed to live in the resources folder, but if you supply a full URL (relative should work too, so basically anything with a / in it will trigger this logic) it will instead link to there.  And of course the tooltip will tell you all of this.

Reply:
If it's evaluated relative to the resources folder, and "../mypuzzle/mycss.css" works, that's sufficient IMO (assuming it's documented).